packages/embark-accounts-manager/src/index.ts

```diff
@@ -0,0 +1,166 @@
+    if (!this.ready) {
+      return callback(null, params);
+    }
+    if (params.reqData.method === "eth_sendTransaction" && this.accounts.length) {
+      // Check if we have that account in our wallet
+      const account = this.accounts.find((acc) => Web3.utils.toChecksumAddress(acc.address) === Web3.utils.toChecksumAddress(params.reqData.params[0].from));
+      if (account && account.privateKey) {
+        return this.signTransactionQueue.push({payload: params.reqData.params[0], account}, (err: any, newPayload: any) => {
+          if (err) {
+            return callback(err, null);
+          }
+          params.reqData.method = "eth_sendRawTransaction";
+          params.reqData.params = [newPayload];
+          callback(err, params);
+        });
+      }
+    }
+    callback(null, params);
+  }
+
+  public arrayEqual(arrayA: string[], arrayB: string[]) {
+    if (arrayA.length !== arrayB.length) {
+      return false;
+    } else {
+      return arrayA.every((address, index) => Web3.utils.toChecksumAddress(address) === Web3.utils.toChecksumAddress(arrayB[index]));
+    }
+  }
+
+  private async checkBlockchainResponse(params: any, callback: (error: any, result: any) => void) {
+    if (!this.ready) {
+      return callback(null, params);
+    }
+    if ((params.reqData.method === "eth_accounts" || params.reqData.method === "personal_listAccounts") && this.accounts.length) {
+      if (!this.arrayEqual(params.respData.result, this.nodeAccounts)) {
+        this.nodeAccounts = params.respData.result;
+        const web3 = await this.web3;
+        this.accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, this.nodeAccounts);
+      }
+      params.respData.result = this.accounts.map((acc) => {
+        if (acc.address) {
+          return acc.address;
+        }
+        return acc;
+      });
+      return callback(null, params);
+    }
+    callback(null, params);
+  }
+
+  private setReady() {
+    if (this.ready) {
+      return;
+    }
+    this.ready = true;
+    this.events.emit("accounts-manager:ready");
+  }
+
+  private async parseAndFundAccounts() {
+    const web3 = await this.web3;
+
+    const nodeAccounts = await web3.eth.getAccounts();
+    this.nodeAccounts = nodeAccounts;
+    this.accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, nodeAccounts);
+
+    if (!this.accounts.length || !this.embark.config.blockchainConfig.isDev) {
+      return this.setReady();
+    }
+    async.eachLimit(this.accounts, 1, (account, eachCb) => {
+      if (!account.address) {
+        return eachCb();
+      }
+      fundAccount(web3, account.address, account.hexBalance, eachCb);
+    }, (err) => {
+      if (err) {
+        this.logger.error(__("Error funding accounts"), err.message || err);
+      }
+      this.setReady();
```

Comment:
Could we please change this to 
```
this.ready = true;
```
because we don't need a function for this, as we only need to modify a private variable.